rest/rest-api/src/main/java/org/ow2/proactive_grid_cloud_portal/common/SchedulerRestInterface.java

```diff
@@ -1198,6 +1166,28 @@ String loginOrRenewSession(@HeaderParam("sessionid") String sessionId,
             @FormParam("username") String username, @FormParam("password") String password)
                     throws SchedulerRestException, LoginException, NotConnectedRestException;
 
+
+    /**
+     * Renew the session identified by the given {@code sessionId} if it exists or create a new session.
+     * <p>
+     * login to the scheduler using a multipart form
+     * can be used either by submitting
+     * - 2 fields: username and password
+     * - a credential file with field name 'credential'
+     *
+     * @param multipart password
+     * @param sessionId session id identifying a session to renew.
+     * @return the new session id to use.
+     * @throws SchedulerRestException
+     */
+    @PUT
+    @Path("session")
+    @Consumes(MediaType.MULTIPART_FORM_DATA)
+    @Produces("application/json")
+    String loginOrRenewSession(@HeaderParam("sessionid") String sessionId,
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Refactor this method to throw at most one checked exception instead of: java.security.KeyException, org.ow2.proactive_grid_cloud_portal.scheduler.exception.SchedulerRestException, javax.security.auth.login.LoginException, org.ow2.proactive_grid_cloud_portal.scheduler.exception.NotConnectedRestException [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS1160)